rfi_file_monitor/applicationwindow.py

```diff
@@ -425,12 +433,52 @@ def __init__(self, *args, **kwargs):
             str, # error message
         )
 
-        files_scrolled_window = Gtk.ScrolledWindow(
-            halign=Gtk.Align.FILL, valign=Gtk.Align.FILL,
-            hexpand=True, vexpand=True)
-        output_frame.add(files_scrolled_window)
+        self._files_tree_model_filter = Gtk.TreeModelFilter(child_model=self._files_tree_model)
+        self._files_tree_model_filter.set_visible_func(self._files_tree_model_visible_func)
+
+        output_grid = Gtk.Grid(**EXPAND_AND_FILL)
+
+        filters_grid = Gtk.Grid(border_width=5,
+            halign=Gtk.Align.FILL, valign=Gtk.Align.CENTER,
+            hexpand=True, vexpand=False,
+            column_spacing=5,
+        )
+        output_grid.attach(filters_grid, 0, 0, 1, 1)
+
+        label = Gtk.Label(label='Table Filters:',
+            halign=Gtk.Align.START, valign=Gtk.Align.CENTER,
+            hexpand=False, vexpand=False,
+        )
+        filters_grid.attach(label, 0, 0, 1, 1)
+        state_filters_button = Gtk.Button(label='Status',
+            halign=Gtk.Align.START, valign=Gtk.Align.CENTER,
+            hexpand=False, vexpand=False,
+        )
+        filters_grid.attach(state_filters_button, 1, 0, 1, 1)
+
+        state_filter_popover = Gtk.Popover.new_from_model(state_filters_button, self.props.application.filter_popover_menu)
+        state_filters_button.connect('clicked', self._state_filters_button_clicked, state_filter_popover)
+
+        label = Gtk.Label(label='Name:',
+            halign=Gtk.Align.START, valign=Gtk.Align.CENTER,
+            hexpand=False, vexpand=False,
+        )
+        filters_grid.attach(label, 2, 0, 1, 1)
+        self._name_filter_entry = Gtk.Entry(
+            placeholder_text='Wildcards are allowed',
```

Comment:
maybe given a example 'e.g. file_*' not sure how many people know wild cards

Reply:
I will use the same text you used